framework/platform/bpl/uci/bpl_uci.cpp

```diff
@@ -0,0 +1,263 @@
+#include "bpl/bpl_uci.h"
+
+#include <mapf/common/logger.h>
+#include <mapf/common/utils.h>
+
+extern "C" {
+#include <uci.h>
+}
+
+bool uci_entry_exists(const std::string &config_file, const std::string &entry_type,
+                      const std::string &entry_name)
+{
+    //entry: config_file: entry_type(entry_name)
+    LOG(TRACE) << "entry: " << config_file << ": " << entry_type << "(" << entry_name << ")";
+
+    struct uci_context *ctx = nullptr;
+    if (!(ctx = uci_alloc_context())) {
+        LOG(ERROR) << "uci context allocate failed!";
+        return false;
+    }
```

Comment:
Since this code repeats in every function, it should probably be extracted into a local module function.
Also consider wrapper it in a `shared_ptr` with a custom destructor to automatically free the context.

Something like this:
```c++
std::shared_ptr<uci_context> alloc_context() {
    auto ctx = std::shared_ptr<uci_context>(uci_alloc_context(), [](uci_context *ctx) {
        if (ctx) {
            uci_free_context(ctx);
        }
    });

    LOG_IF(!ctx, ERROR) << "Failed allocating UCI context!";
    return ctx;
}
```

Then, all you have to do in every function is:
```c++
auto ctx = alloc_context();
if (!ctx) {
    return false;
}
```

And no need to worry about freeing the context.

Reply:
Added `shared_ptr` for the context